sktime/transformations/series/peak.py

```diff
@@ -0,0 +1,450 @@
+        The * can be one of the following:
+        * Y - year
+        * Q - quarter
+        * M - month
+        * W - week
+        * D - day
+        * H - hour
+
+    working_[^]_start, working_[^]_end : list,
+        working start and working end should be specified in form of:
+        working_[*]_start = [start1, start2, ...],
+        working_[*]_end = [end1, end2, ...].
+        E.g. 1, one working interval: working_hour_start=[8], working_hour_end=[16]
+        means we have just ONW working hour interval where work starts at 8 am and
+        work ends at 16. E.g. 2, two working intervals: working_hour_start=[8, 15],
+        working_hour_end=[12, 19] means we have TWO working hour intervals where the
+        first starts at 8 am and ends at 15. The second  starts at 15 am and ends at 19.
+        - we may have more than TWO intervals.
+        The ^ can be one of the following:
+        ^ H - hour
+
+    keep_original_columns :  boolean, optional, default=False
+        Keep original columns in X passed to `.transform()`.
+    keep_original_PeakTimeFeature_columns: boolean, optional, default=False
+        Keep original PeakTimeFeature dataframe columns including all separate
+        peak/working columns, e.g., peak_hour_1, peak_hour_2, peak_week_1,
+        peak_week_2, ...
+
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.peak import PeakTimeFeature
+    >>> from sktime.datasets import load_solar
+    >>> y = load_solar()
+    >>> y = y.tz_localize(None)
+    >>> y = y.asfreq("H")
+
+    Example ONE interval for peak hour and working hour
+    Returns columns is_peak_hour, is_working_hour (based on one start/end interval)
+
+    >>> transformer = PeakTimeFeature(ts_freq="H",
+    ... peak_hour_start=[6], peak_hour_end=[9],
+    ... working_hour_start=[8], working_hour_end=[16]
+    ... )
+    >>> y_hat_peak = transformer.fit_transform(y)
+    >>> y_hat_peak
+
+
+    Example TWO intervals for peak hour and  working hour
+    Returns columns is_peak_hour, is_working_hour (based on two start/end intervals)
+
+    >>> transformer = PeakTimeFeature(ts_freq="H",
+    ... peak_hour_start=[6, 16], peak_hour_end=[9, 20],
+    ... working_hour_start=[8, 15], working_hour_end=[12, 19]
+    ... )
+    >>> y_hat_peak = transformer.fit_transform(y)
+    >>> y_hat_peak
+
+
+    Example TWO intervals, We may have peak for different seasonality
+    Here an example for peak hour, peak day, peak week, peak month
+    Returns columns is_peak_hour, is_peak_day, is_peak_week, is_peak_month
+    (based on two start/end intervals)
+
+    >>> transformer = PeakTimeFeature(ts_freq="H",
+    ... peak_hour_start=[6, 16], peak_hour_end=[9, 20],
+    ... peak_day_start=[1, 2], peak_day_end=[2, 3],
+    ... peak_week_start=[35, 45], peak_week_end=[40, 52],
+    ... peak_month_start=[1, 7], peak_month_end=[6, 12]
+    ... )
+    >>> y_hat_peak = transformer.fit_transform(y)
+    >>> y_hat_peak
+
```

Comment:
Since this line is empty, doctest will interpret this as your last line will return nothing as output. But that's not the case, and so this is going to fail.

I believe a `pandas.DataFrame` is the oitput here? That may be tricky to use in doctest, so you have couple of easy workarounds:

1. Print a property of `y_hat_peak`, for example shape or columns or something similar instead of the raw object itself.
2. Skip the doctest for this particular line, which is probably same as removing this altogether from the docstring.

Reply:
Thanks a lot @yarnabrina and @fkiraly for suggestions and comments!
All checks have passed